operations/kubernetes/csm-service-upgrade-FAILED.md

```diff
@@ -0,0 +1,193 @@
+job.batch/cray-dns-unbound-manager-1650322800 condition met
+job.batch/cray-dns-unbound-manager-1650322980 condition met
+job.batch/cray-dns-unbound-manager-1650323160 condition met
+timed out waiting for the condition on jobs/cray-dns-unbound-manager-1650323280
+timed out waiting for the condition on jobs/cray-dns-unbound-manager-1650324240
+[ERROR] - Unexpected errors, check output above
+/tmp
+CSM Service upgrade failed after 3 retries
+```
+
+
+## Solution
+
+The cause of this failure is the unbound-manager job failing, it can be checking looking to the pods.
+Look for the failed unbound pod.
+
+```sh
+ncn-m001:~ # kubectl get pod -A | grep unbound
+services            cray-dns-unbound-68cb6d759b-2r68h                                 2/2     Running            0          5d22h
+services            cray-dns-unbound-68cb6d759b-9vxbk                                 2/2     Running            0          4d15h
+services            cray-dns-unbound-68cb6d759b-x7j6d                                 2/2     Running            0          6d17h
+services            cray-dns-unbound-7d845ddf8b-4dgcl                                 1/2     Running            0          14h
+services            cray-dns-unbound-7d845ddf8b-r6nxt                                 1/2     Running            0          14h
+services            cray-dns-unbound-7d845ddf8b-w4hgf                                 1/2     Running            0          14h
+services            cray-dns-unbound-coredns-4t45h                                    0/2     Completed          0          13h
+services            cray-dns-unbound-manager-1650322800-vn26p                         0/2     Completed          0          14h
+services            cray-dns-unbound-manager-1650322980-hhfsr                         0/2     Completed          0          14h
+services            cray-dns-unbound-manager-1650323160-lsts5                         0/2     Completed          0          14h
+services            cray-dns-unbound-manager-1650373560-47fxg                         0/2     Completed          0          14m
+services            cray-dns-unbound-manager-1650373560-4zsv7                         0/2     Completed          0          11m
+services            cray-dns-unbound-manager-1650373560-8cpq6                         0/2     Completed          0          19m
+services            cray-dns-unbound-manager-1650373560-977xb                         0/2     Completed          0          7m40s
+services            cray-dns-unbound-manager-1650373560-wsnl5                         0/2     Completed          0          16m
+services            cray-dns-unbound-manager-1650374520-mq4wk                         0/2     Completed          0          4m57s
+services            cray-dns-unbound-manager-1650374520-tdhsl                         1/2     Error              0          2m14s
+```
+
+Continuing the troubleshootng and looking for most recent log we can understand that it can't talk to Kea.
```

Comment:
2. Look for the most recent log to verify if Unbound is able talk to Kea.

    ```sh
    ncn-m001# kubectl -n services logs cray-dns-unbound-manager-1650374520-tdhsl -c manager
    2022-04-19 13:24:53,530 - __main__ - INFO - Querying Kea in the cluster to find any updated records we need to set
    Traceback (most recent call last):
    File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
        resp = conn.urlopen(
    File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
        return self.urlopen(
    File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
        return self.urlopen(
    File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 846, in urlopen
        return self.urlopen(
    [Previous line repeated 7 more times]
    File "/usr/lib/python3.8/site-packages/urllib3/connectionpool.py", line 836, in urlopen
        retries = retries.increment(method, url, response=response, _pool=self)
    File "/usr/lib/python3.8/site-packages/urllib3/util/retry.py", line 573, in increment
        raise MaxRetryError(_pool, url, error or ResponseError(cause))
    urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
    During handling of the above exception, another exception occurred:
    Traceback (most recent call last):
    File "/srv/unbound/manager.py", line 573, in <module>
        main()
    File "/srv/unbound/manager.py", line 167, in main
        resp = kea_api('POST', '/', headers=kea_headers, json=kea_request)
    File "/srv/unbound/manager.py", line 83, in __call__
        response = http.request(method=method, url=url, headers=headers, **kwargs)
    File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
        resp = self.send(prep, **send_kwargs)
    File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
        r = adapter.send(request, **kwargs)
    File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 507, in send
        raise RetryError(e, request=request)
    requests.exceptions.RetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
    ```

    In this example, Kea is also in bad state and is reporting `"failed to select a subnet...."` messages, meaning it has not loaded the configuration.